src/App.jsx

```diff
@@ -1,12 +1,25 @@
-import React, { useState } from 'react';
-import Data from './Data.json';
+import React, { useState, useEffect } from 'react';
+import data from './data.json';
 
 import Header from './Header';
 import ToDoList from './ToDoList';
 import ToDoForm from './ToDoForm';
 
 function App() {
-    const [toDoList, setToDoList] = useState(Data);
+    const [toDoList, setToDoList] = useState([]);
+
+    useEffect(() => {
```

Comment:
awesome. just move the `fetchData` function outside of  the `useEffect` hook.
and bonus task - try to show a loader in the UI while the data is fetching (while developing the feature you can mock the loading time by awaiting an empty promise with setTimeout)